end_to_end_tests/baseline_openapi_3.0.json

```diff
@@ -2841,15 +2841,17 @@
           "modelType": {
             "type": "string"
           }
-        }
+        },
+        "required": ["modelType"]
```

Comment:
This is necessary in order for the discriminator in the spec to really be valid: a discriminator property must be a required property in all of the variant schemas. My current implementation wouldn't actually catch a mistake like this, but I figured it was best to have the test spec be valid.

Reply:
At least, that's my reading of [what OpenAPI says](https://spec.openapis.org/oas/v3.1.0.html#discriminator-object). I'm basing it on the statement "The expectation now is that a property with name petType MUST be present in the response payload, and the value will correspond to the name of a schema defined in the OAS document."

However, this is a case (one of many) of the OpenAPI specification using a mix of normative and non-normative language. The MUST seems unambiguous, but it's in a paragraph that's describing this specific example with types of pets, where the schemas _do_ say the property is required. So do they mean a discriminator property must always be required, or just that that's the behavior this example is illustrating?

Logically I feel like it makes sense for it to be required, and if it isn't, then I'm not sure what the client behavior should be: (a) refuse to parse the object if the property is missing, or (b) just fall back to "try parsing it as each of these types until one of them works"? What I've implemented so far is (a).